ctapipe/reco/shower_processor.py

```diff
@@ -0,0 +1,193 @@
+        ],
+        help=QualityQuery.quality_criteria.help,
+    ).tag(config=True)
+
+
+class ShowerProcessor(Component):
+    """
+    Takes DL1/parameters data and estimates the shower geometry.
+    Should be run after ImageProcessor to produce all DL1b information.
+
+    For now the only supported reconstructor is HillasReconstructor.
+    """
+
+    def __init__(
+        self,
+        subarray: SubarrayDescription,
+        is_simulation,
+        reconstruct_energy,
+        classify,
+        config=None,
+        parent=None,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        subarray: SubarrayDescription
+            Description of the subarray. Provides information about the
+            camera which are useful in calibration. Also required for
+            configuring the TelescopeParameter traitlets.
+        is_simulation: bool
+            If true, also process simulated images if they exist
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent: ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+        """
+
+        super().__init__(config=config, parent=parent, **kwargs)
+        self.subarray = subarray
+        self.check_shower = ShowerQualityQuery(parent=self)
+        self._is_simulation = is_simulation
+        self.reconstruct_energy = reconstruct_energy
+        self.classify = classify
+        self.reconstructor = HillasReconstructor()
+
+    def _reconstruct_shower(
+        self,
+        event,
+        default=DEFAULT_SHOWER_PARAMETERS,
+    ) -> ReconstructedShowerContainer:
+        """Perform shower reconstruction.
+
+        Parameters
+        ----------
+        tel_id: int
+            which telescope is being cleaned
+        image: np.ndarray
+            image to process
+        signal_pixels: np.ndarray[bool]
+            image mask
+        peak_time: np.ndarray
+            peak time image
+        Returns
+        -------
+        ReconstructedShowerContainer:
+            direction in the sky with uncertainty,
+            core position on the ground with uncertainty,
+            h_max with uncertainty,
+            is_valid boolean for successfull reconstruction,
+            average intensity of the intensities used for reconstruction,
+            measure of algorithm success (if fit),
+            list of tel_ids used if stereo, or None if Mono
+        """
+
+        # Read only valid HillasContainers (minimum condition to continue)
+        hillas_dict = {
+            tel_id: dl1.parameters.hillas
+            for tel_id, dl1 in event.dl1.tel.items()
+            if np.isfinite(event.dl1.tel[tel_id].parameters.hillas.intensity)
```

Comment:
Why not use the shower_criteria already here?

Reply:
Yes, technically the default shower criteria should also check for this, should we just drop this then?

Reply:
No, you have to here use it! See comment below.

Reply:
done